lib/shared/bucketing-assembly-script/__tests__/bucketing/segmentation.test.ts

```diff
@@ -394,6 +395,184 @@ describe('SegmentationManager Unit Test', () => {
+                        _audiences: ['test']
+                    }
+                ],
+                operator: 'and'
+            }
+            it('should pass seg for an AND operator', () => {
+
+                const audiences = {
+                    'test': {
+                        _id: 'test',
+                        filters: operator
+                    }
+                }
+                assert.strictEqual(true, evaluateOperator(
+                    { data, operator: audienceMatchOperator, audiences }))
+            })
+            it('should not pass seg when not in audience for an AND operator', () => {
+
+                const audiences = {
+                    'test': {
+                        _id: 'test',
+                        filters: operator
+                    }
+                }
+
+                assert.strictEqual(false, evaluateOperator(
+                    { data, operator: audienceMatchOperatorNotEqual, audiences }))
+            })
+            it('should pass seg for nested audiences', () => {
+                const nestedAudienceMatchOperator = {
+                    filters: [
+                        {
+                            type: 'audienceMatch',
+                            comparator: '=',
+                            _audiences: ['nested']
+                        }
+                    ],
+                    operator: 'and'
+                }
+
+                const audiences = {
+                    'test': {
+                        _id: 'test',
+                        filters: operator
+                    },
+                    'nested': {
+                        _id: 'nested',
+                        filters: audienceMatchOperator
+                    }
+                }
+                assert.strictEqual(true, evaluateOperator(
+                    { data, operator: nestedAudienceMatchOperator, audiences }))
+            })
+            it('should not pass seg for nested audiences with !=', () => {
+                const nestedAudienceMatchOperator = {
+                    filters: [
+                        {
+                            type: 'audienceMatch',
+                            comparator: '!=',
+                            _audiences: ['nested']
+                        }
+                    ],
+                    operator: 'and'
+                }
+
+                const audiences = {
+                    'test': {
+                        _id: 'test',
+                        filters: operator
+                    },
+                    'nested': {
+                        _id: 'nested',
+                        filters: audienceMatchOperator
+                    }
+                }
+                assert.strictEqual(false, evaluateOperator(
+                    { data, operator: nestedAudienceMatchOperator, audiences }))
+            })
+            it('should pass seg for an AND operator with multiple values', () => {
```

Comment:
i dont think the and operator is really being tested in this test, since it applies to the "filters" array and all your filters arrays only have a length of 1

Reply:
yeah I will rename this test. Its just titled like this because AND operator is the default. I dont think we need to re-write tests for the foundational segmentation code for every new filter type. The code that handles the AND level filtering hasnt changed at all, so as long as the code that actually tests the `audieceMatch` filter type works then we know that AND filtering on it will also work.

Reply:
well don't we need to test that audiences are correctly passed down?

Reply:
The AND operator functionality is contained inside the function that all these tests are calling, so that's already tested. Plus it will be double covered once I add a test to the bucketing file that calls the parent function of that

Reply:
oh sorry i thought this was in reference to the other comment about testing at a higher level